File: app/src/main/java/me/zhanghai/android/files/compat/TaskDescriptionCompat.java
/*
 * Copyright (c) 2019 Hai Zhang <dreaming.in.code.zh@gmail.com>
 * All Rights Reserved.
 */

package me.zhanghai.android.files.compat;

import android.app.ActivityManager;

import androidx.annotation.ColorInt;
import androidx.annotation.NonNull;
import me.zhanghai.java.reflected.ReflectedMethod;

public class TaskDescriptionCompat {

    static {
        RestrictedHiddenApiAccess.allow();
    }

    @RestrictedHiddenApi
    private static final ReflectedMethod<ActivityManager.TaskDescription> sSetPrimaryColorMethod =
            new ReflectedMethod<>(ActivityManager.TaskDescription.class, "setPrimaryColor",
                    int.class);

    private TaskDescriptionCompat() {}

    public static void setPrimaryColor(@NonNull ActivityManager.TaskDescription taskDescription,
                                       @ColorInt int primaryColor) {
        sSetPrimaryColorMethod.invoke(taskDescription, primaryColor);
    }
}

File: app/src/main/java/me/zhanghai/android/files/compat/TaskDescriptionCompat.kt
/*
 * Copyright (c) 2019 Hai Zhang <dreaming.in.code.zh@gmail.com>
 * All Rights Reserved.
 */

package me.zhanghai.android.files.compat

import android.app.ActivityManager.TaskDescription
import androidx.annotation.ColorInt
import me.zhanghai.java.reflected.ReflectedMethod

@RestrictedHiddenApi
private val setPrimaryColorMethod =
    ReflectedMethod(TaskDescription::class.java, "setPrimaryColor", Int::class.java)

fun TaskDescription.setPrimaryColorCompat(@ColorInt primaryColor: Int) {
    setPrimaryColorMethod.invoke<Unit>(this, primaryColor)
}
